x-template.js

```diff
@@ -23,6 +21,12 @@ class TemplateEngine {
   static #STATE = Symbol();
   static #ARRAY_STATE = Symbol();
 
+  // It’s more performant to clone a single fragment, so we keep a reference.
+  static #fragment = new DocumentFragment();
```

Comment:
Note that the creation of the document fragment root is moved to “x-template.js” now. And, there’s no need to conditionally create it as “x-template” can safely assume that it will always be run in a browser.